python/cugraph/cugraph/sampling/heterogeneous_neighbor_sample.py

```diff
@@ -0,0 +1,349 @@
+# Copyright (c) 2022-2025, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cudf
+from pylibcugraph import ResourceHandle
+from pylibcugraph import (
+    heterogeneous_uniform_neighbor_sample as pylibcugraph_heterogeneous_uniform_neighbor_sample,
+    heterogeneous_biased_neighbor_sample as pylibcugraph_heterogeneous_biased_neighbor_sample,
+)
+from cugraph.sampling.sampling_utilities import sampling_results_from_cupy_array_dict
+
+from cugraph.structure import Graph
+
+
+import numpy
+
+import cudf
+import cupy as cp
+import warnings
+
+from typing import Union, Tuple, Sequence, List
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from cugraph import Graph
+
+
+# FIXME: Move this function to the utility module so that it can be
+# shared by other algos
+def ensure_valid_dtype(input_graph, start_list):
+    vertex_dtype = input_graph.edgelist.edgelist_df.dtypes.iloc[0]
+    if isinstance(start_list, cudf.Series):
+        start_list_dtypes = start_list.dtype
+    else:
+        start_list_dtypes = start_list.dtypes.iloc[0]
+
+    if start_list_dtypes != vertex_dtype:
+        warning_msg = (
+            "Uniform neighbor sample requires 'start_list' to match the graph's "
+            f"'vertex' type. input graph's vertex type is: {vertex_dtype} and got "
+            f"'start_list' of type: {start_list_dtypes}."
+        )
+        warnings.warn(warning_msg, UserWarning)
+        start_list = start_list.astype(vertex_dtype)
+
+    return start_list
+
+
+def heterogeneous_uniform_neighbor_sample(
```

Comment:
Should this still have `_uniform` in the function name?

Reply:
Right this has been resolved